src/state/search.ts

```diff
@@ -0,0 +1,113 @@
+import { atom, DefaultValue, selector } from "recoil";
+import {
+  CatalogSearchParams,
+  ExtendedCatalogPackage,
+} from "../api/catalog-search";
+import { CatalogSearchSort } from "../api/catalog-search/constants";
+import { CDKType } from "../constants/constructs";
+import { Language } from "../constants/languages";
+
+export const queryState = atom<string>({
+  key: "search-queryState",
+  default: "",
+});
+
+export const keywordsState = atom<string[]>({
+  key: "search-keywordsState",
+  default: [],
+});
+
+export const cdkTypeState = atom<CDKType | undefined>({
+  key: "search-cdkTypeState",
+  default: undefined,
+});
+
+export const cdkMajorState = atom<number | undefined>({
+  key: "search-cdkMajorState",
+  default: undefined,
+});
+
+export const languagesState = atom<Language[]>({
+  key: "search-languagesState",
+  default: [],
+});
+
+export const offsetState = atom<number>({
+  key: "search-offsetState",
+  default: 0,
+});
+
+export const sortState = atom<CatalogSearchSort | undefined>({
+  key: "search-sortState",
+  default: undefined,
+});
+
+export const resultsState = atom<ExtendedCatalogPackage[]>({
+  key: "search-resultsState",
+  default: [],
+});
+
+export const pageLimitState = atom<number>({
+  key: "search-pageLimitState",
+  default: 0,
+});
+
+export const pageState = atom<ExtendedCatalogPackage[]>({
+  key: "search-pageState",
+  default: [],
+});
+
+export const tagsState = atom<string[]>({
+  key: "search-tagsState",
+  default: [],
+});
```

Comment:
As discussed briefly separately, most of these are duplicated from the url query parameters. In something like redux in the past, I would have argued against manually synchronizing these to the store and instead pulling them from the URL, and dispatching actions to change the URL directly to ensure a single source of truth and a one way data flow.

That being said, is it possible to only have a single atom for `resultsState` that is the data from catalog.json, then the selector takes that and the query params as input to compute the filtered results to be displayed, and components dispatch changes directly to the query params that they need to in order to change results?

Reply:
This might make the most sense to do as a follow-up item. I plan on migrating our data providers to atoms rather than react-contexts, which would pave the way to the work you're suggesting.